### PR TITLE
Some changes to make various parts more robust to errors (WorkflowUsi…

### PR DESCRIPTION
**Pull request by vinjana**
…ngMergedBams). getAllBamFiles() in BasicCOProjectRuntimeService that is ignorant of tumor-control concept (not like the WorkflowUsingMergedBam, which only knows tumor and control BAMs).

### CODE REVIEW

File: src/de/dkfz/b080/co/common/BasicCOProjectsRuntimeService.groovy [hunk 1]
```diff
@@ -151,7 +152,7 @@ class BasicCOProjectsRuntimeService extends RuntimeService {
 
     static List<Sample> extractSamplesFromFilenames(List<File> filesInDirectory, ExecutionContext context) {
         COConfig cfg = new COConfig(context)
-        LinkedList<Sample> samples = [];
+        LinkedList<Sample> samples = [] as LinkedList;
```
Comment (dankwart-de) [inline src/de/dkfz/b080/co/common/BasicCOProjectsRuntimeService.groovy]: Idea?

File: src/de/dkfz/b080/co/common/BasicCOProjectsRuntimeService.groovy [hunk 2]
```diff
@@ -191,8 +226,7 @@ class BasicCOProjectsRuntimeService extends RuntimeService {
         } else if (cfg.extractSamplesFromBamList) {
             List<File> bamFiles = cfg.getBamList().collect { String f -> new File(f); }
             samples = extractSamplesFromFilenames(bamFiles, context)
-            // @Michael: Should that not better be called "bam_list" in analogy to "fastq_list"?
-            extractedFrom = "bamfile_list configuration value "
+            extractedFrom = "${COConstants.CVALUE_BAMFILE_LIST} configuration value "
```
Comment (dankwart-de) [inline src/de/dkfz/b080/co/common/BasicCOProjectsRuntimeService.groovy]: extracted from configuration value?

File: src/de/dkfz/b080/co/common/BasicCOProjectsRuntimeService.groovy (hunk 1 above)
Comment (vinjana) [reply src/de/dkfz/b080/co/common/BasicCOProjectsRuntimeService.groovy]: yes

File: src/de/dkfz/b080/co/common/BasicCOProjectsRuntimeService.groovy (hunk 2 above)
Comment (vinjana) [reply src/de/dkfz/b080/co/common/BasicCOProjectsRuntimeService.groovy]: Yes. Why?

**Pull request merged**